app/models/course/settings/materials_component.rb

```diff
@@ -6,9 +6,10 @@ class Course::Settings::MaterialsComponent
 
   # Initialises the settings adapter
   #
-  # @param [#settings] settings The settings object provided by the settings_on_rails gem.
-  def initialize(settings)
-    @settings = settings
+  # @param [#current_course, #key] context The component that is the context for the settings
+  def initialize(context)
+    @context = context
```

Comment:
To be consitent with the component framework, better to use `SimpleDelegator` to delegate missing methods to context instead.

e.g. https://github.com/Coursemology/coursemology2/blob/master/app/controllers/components/course/achievements_component.rb#L2